### PR TITLE
HD CSS Color Guide

### LINKED ISSUE #4525: [A11y][CSS] High Definition CSS Color Guide [1/10/2023]
* Proposed title: High Definition CSS Color Guide
* Proposed by: argyle
* Related areas: Accessibility,CSS
* Content type: Reference (https://documentation.divio.com/reference/)
* Deadline: 1/10/2023
* Deadline explanation: Chrome 109 release, which will have a majority of these color features hitting stable
* [Draft](https://docs.google.com/document/d/1pMnudZk0NYDKiXnd81Mr_wq5td_EyBge2MYGfZGArDQ/edit)
* Content summary: CSS Color 4 and 5 brings more color spaces, color manipulation functions and overall more colors and gradients to choose from, and this post intends to serve as a canonical guide to the syntax, features, and capabilities of these CSS specs.

### PR DESCRIPTION
**Pull request by argyleink**
Fixes #4525

<br>

Changes proposed in this pull request:
- New article: HD Color Guide
- New blog post to introduce the article: Meet The New CSS Color Spaces
- Update to `color-mix()` post: links to the new HD Color Guide

<br>

To Do:
- [x] get final reviews from DCC editors
- [x] get final reviews from engineers
- [x] update blog post hero image and subtitle
- [x] update dates

### CODE REVIEW

**Comment by argyleink** [comment]
missing gradient hue interpolation CSS examples around shorter, longer etc. only shows the colorjs versions.

**Comment by rachelandrew** [comment]
@argyleink If this is shipping in 110 we need someone to update Chrome Status https://chromestatus.com/feature/6280020765966336

**Comment by argyleink** [comment]
> @argyleink If this is shipping in 110 we need someone to update Chrome Status  

i'm still waiting on confirmation after holidays, will pass on this tip though and report back when I know

**Review by ajul — COMMENTED** [review]
Left some comments. Much is subjective, feel free to take or ignore anything.

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 1]
```diff
@@ -0,0 +1,1458 @@
...
+  - css
+hero: image/vS06HQ1YTsbMKSFTIPl2iogUQP73/lUKgvbcTc1Lg3xNUdCpq.png
+alt: >
+  A moment from The Wizard of Oz movie and half of it is in black and white and the other half is in color.
+---
+
+For over 25 years, `sRGB` (standard red green blue) has been the only color
+gamut for CSS gradients and colors, with color space offerings within it like
+`rgb()`, `hsl` and hex. It is the most common color gamut capability amongst
+displays; a common denominator.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:21]: coo, will update

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 2]
```diff
@@ -0,0 +1,1458 @@
...
+offerings found in images and video.
+[Trapped](https://www.youtube.com/watch?v=_z3Gpk5dbCk), only showing [30% of the
+colors the human eye can
+see](https://en.wikipedia.org/wiki/Wide-gamut_RGB_color_space), when modern
+alternatives offer 70% or more. We can now thank [Color Level
+4](https://www.w3.org/TR/css-color-4) for helping us escape this trap.
+
+Shipped in Chrome 110 is support for [CSS Color
+4](https://www.w3.org/TR/css-color-4/) gamuts and color spaces, joining Safari
+who's had support for `display-p3` since 2016. CSS can now support HD (high
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:35]: sgtm, updating

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 3]
```diff
@@ -0,0 +1,1458 @@
...
+In addition to more colors, [arguably the most vivid colors the display is
+capable
+of](https://lea.verou.me/2020/04/lch-colors-in-css-what-why-and-how/#:~:text=most%20vivid%20colors%20the%20screen%20can%20display),
+additional color spaces provide new tools and methods for managing and creating
+color systems. HSL has a "lightness" channel, which was the best web developers
+had in CSS, until LCH's "perceptual lightness."
+
+In the following example, `50%` lightness in HSL is compared to `50%` in LCH,
+across a series of hues. The gray next to each color swatch is the perceived
+lightness as defined in LCH. Notice how HSL is wildly sporadic while LCH remains
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:65]: done.

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 4]
```diff
@@ -0,0 +1,1458 @@
...
+  {% Video src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/swYaLIEXuDRZ2VO8SCLH.mp4" %}
+
+  <figcaption>
+    A series of images are shown transitioning between wide and narrow color
+    gamuts, illustrating color vividness and its effects.<br>
+    <a href="https://ciechanow.ski/color-spaces/#:~:text=you%20can%20drag%20the%20slider%20to%20see%20how%20the%20extent%20of%20the%20chromaticity%20triangle%20corresponds%20to%20the%20representable%20colors.">Source</a>
+  </figcaption>
+</figure>
+
+In addition to more colors, [arguably the most vivid colors the display is
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:56]: done.

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 5]
```diff
@@ -0,0 +1,1458 @@
...
+## Overview
+
+The current problem with color and the web, is that the displays most folks have
+in their pockets, laps or mounted on walls are wide gamut, high definition color
+ready. CSS on the other hand, is not high definition ready, until now. The color
+capability of displays grew faster than CSS, now CSS is here to catch up.
+
+There's much more than just "more colors" too. By the end of this article you'll
+be able to specify more colors, enhance gradients, and pick color spaces and
+color gamuts for a task they excel at.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:125]: done.

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 6]
```diff
@@ -0,0 +1,1458 @@
...
+A color space is a mapping of colors where a color gamut is a range of colors.
+Consider a color gamut as a total of particles and a color space as a bottle
+made to hold that range of particles.
+
+Here's an interactive visual of what I mean. Point, drag and zoom around in this
+demo of an RGB cube filled with color particles.
+
+{% Codepen { user: 'web-dot-dev', id: 'gOjrLwm' } %}
+
+Use color gamuts to talk about a range of colors, like low range gamut vs high
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:219]: done. will have both

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 7]
```diff
@@ -0,0 +1,1458 @@
...
+{% Aside %}
+Colors are in the `sRGB Gamut`; 35.9% of the visible spectrum.
+{% endAside %}
+
+<!-- {% BrowserCompat 'css.types.color.hsl' %} -->
+
+One of the first color spaces to orient itself towards human language and
+communication, HSL (hue saturation and lightness) offers all the colors in the
+sRGB gamut while not requiring your brain to know how red, green and blue
+interact. Like RGB, it also originally had commas in the syntax, but moving
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:329]: thx, done.

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 8]
```diff
@@ -0,0 +1,1458 @@
...
+specification has instructions for the browsers on how to handle these cross
+color space interpolations. In the above case, browsers should look at the first
+color's color space for the gradient's color space. This is called the [host
+syntax](https://www.w3.org/TR/css-color-4/#host-syntax), the first color becomes
+the host and its syntax determines the space.
+
+##### Less banding thanks to 16-bit color
+
+Until now in Chrome, all colors were saved in 8-bit memory, but in order to
+support HD color these needed to be upgraded to 16-bit. This increases the
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:1001]: hm yes, i dont really talk about ints vs floats at all. lemme think on that thanks!

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 9]
```diff
@@ -0,0 +1,1458 @@
...
+
+1. **Progressive enhancement**<br>Use `@supports` and `@media` to assess the
+   capabilities of the user's browser, and if conditions are met, provide wide
+   gamut colors.
+
+If a browser doesn't understand `display-p3` color:
+
+```css//1
+color: red;
+color: color(display-3 1 0 0);
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:1311]: yeah, what's the issue / suggestion?

[the remark this answers is not in the document]
Comment (ajul) [reply site/en/blog/high-definition-css-color-guide/index.md:1311]: The "p".is missing.

[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:1311]: ahh! nice =)
viva peer reviews!

[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:1311]:
```suggestion
color: color(display-p3 1 0 0);
```

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 10]
```diff
@@ -0,0 +1,1458 @@
...
+```
+
+If a browser does understand `display-p3` color:
+
+```css//0
+color: red;
+color: color(display-3 1 0 0);
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:1318]:
```suggestion
color: color(display-p3 1 0 0);
```

**Comment by argyleink** [comment]
Chrome 110 confirmed. Chrome status update request made.

**Comment by argyleink** [comment]
@rachelandrew here's the [Chrome Status their working with](https://chromestatus.com/feature/5147752829288448). they're going to merge / delete one so there's less confusion.

**Comment by rachelandrew** [comment]
@argyleink there is a separate one for color-mix() as well... https://chromestatus.com/feature/5659115386830848 these probably need disentangling.

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 11]
```diff
@@ -0,0 +1,1511 @@
...
+#### Putting the hardware and parsing checks together
+
+While waiting for each browser to implement these new color features, it's a
+good idea to check for both hardware capability and color parsing capability.
+This is often what I use when progressively enhancing colors to high definition:
+
+```css
+:root {
+  --neon-red: red;
+  --neon-blue: blue;
```
Comment (argyleink) [inline site/en/blog/high-definition-css-color-guide/index.md:1467]:
```suggestion
  --neon-red: rgb(100% 0 0);
  --neon-blue: rgb(0 0 100%);
```

**Review by mysteryDate — COMMENTED** [review]
A good start! I have many thoughts. I'll let you get to these before I move on to the rest of the document.

**Comment by rachelandrew** [comment]
@argyleink it looks like this should have had more review in the doc? Can we take it back there until people are happy with it and you've got all your LGTMs?

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 12]
```diff
@@ -0,0 +1,1511 @@
...
+    Preview for yourself: <a href="https://codepen.io/web-dot-dev/pen/poZgXxy">
+    https://codepen.io/web-dot-dev/pen/poZgXxy</a>
+  </figcaption>
+</figure>
+
+In supporting browsers, there's 50% more colors to pick from! You thought 16
+million colors sounded like a lot, wait until you see how many colors some of
+these new spaces can show. Also, think about all those gradients that
+[banded](https://www.willgibbons.com/color-banding/) because there weren't
+enough colors in between your request, that's pacified too.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:92]: fixed in doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 13]
```diff
@@ -0,0 +1,1511 @@
...
+</figure>
+
+This guide will cover where color has been, where it is going, and how CSS will
+enable and support web developers to manage color.
+
+## Overview
+
+The current problem with color and the web, is that the displays most folks have
+in their pockets, laps or mounted on walls are wide gamut, high definition color
+ready. CSS on the other hand, is not high definition ready, until now. The color
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:120]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 14]
```diff
@@ -0,0 +1,1511 @@
...
+capability of displays grew faster than CSS, now CSS is here to catch up.
+
+There's much more than just "more colors" too. By the end of this article you'll
+be able to specify more colors, enhance gradients, pick the color space and
+color gamut for a task they excel at.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:125]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 15]
```diff
@@ -0,0 +1,1458 @@
...
+ready. CSS on the other hand, is not high definition ready, until now. The color
+capability of displays grew faster than CSS, now CSS is here to catch up.
+
+There's much more than just "more colors" too. By the end of this article you'll
+be able to specify more colors, enhance gradients, and pick color spaces and
+color gamuts for a task they excel at.
+
+### What is a color gamut?
+
+A gamut represents the size of something. The phrase "millions of colors" is a
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:129]: moved to doc 👍🏻

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 16]
```diff
@@ -0,0 +1,1511 @@
...
+from. In the following image, three gamuts are compared, and the larger the size
+the more colors it offers.
+
+{% Img
+  src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/mM8vJ9grbV1ssmaF0rPH.png",
+  alt="Color gamuts are compared side by side as a triangle shape.
+  sRGB is the smallest and Rec2020 is the largest.",
+  width="800",
+  height="343"
+%}
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:140]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 17]
```diff
@@ -0,0 +1,1511 @@
...
+
+A color gamut can also have a name. Like a basketball versus a baseball or a
+vente coffee cup versus a grande; a name for the size can help people
+communicate. Learning these color gamut names helps you communicate and quickly
+understand a range of colors.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:145]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 18]
```diff
@@ -0,0 +1,1511 @@
...
+
+This article will introduce you to 7 new gamuts, all with wider range than sRGB
+and each with its own quarks and features to assist you in a color task:
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:148]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 19]
```diff
@@ -0,0 +1,1511 @@
...
+
+### What is a color space?
+
+Color spaces are arrangements of a gamut, establishing a shape and a method of
+accessing colors. Usually these are 3D shapes, like cubes or cylinders. This
+color arrangement determines which colors are next to each other and how
+accessing and interpolating colors will work.
+
+RGB is a cube and colors are accessed by specifying 3 coordinates. HSL is a
+cylinder and colors are accessed via a hue angle and 2 coordinates.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:170]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 20]
```diff
@@ -0,0 +1,1511 @@
...
+    alt="A cube filled with many multiple colored dots.",
+    width="384",
+    height="384"
+  %}
+
+  <figcaption>
+    Above is the sRGB gamut of particles fit into an RGB cube color space
+    <a href="https://cran.r-project.org/web/packages/colordistance/vignettes/color-spaces.html">Image source</a>
+  </figcaption>
+</figure>
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:235]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 21]
```diff
@@ -0,0 +1,1511 @@
...
+## How to access more colors, new spaces, and debug results
+
+[CSS Color 4](https://www.w3.org/TR/css-color-4/) outlines a bunch of new
+features and tools for CSS and color. First, a recap of where color was before
+these new features. Then an overview of the new color spaces, syntaxes and
+tools.
+
+The following Codepen shows all the new and old color syntaxes together:
+
+{% Codepen { user: 'web-dot-dev', id: 'LYBNbzp' } %}
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:246]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 22]
```diff
@@ -0,0 +1,1511 @@
...
+
+### A review of the classic color spaces
+
+Since the 2000's, CSS has been able to ask for color with hexadecimal (hex) and
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:250]: fixed in doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 23]
```diff
@@ -0,0 +1,1511 @@
...
+`rgb()`. Around 2010, depending on your browser, CSS could ask for color in
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:251]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 24]
```diff
@@ -0,0 +1,1511 @@
...
+[`hsl()`](https://caniuse.com/mdn-css_types_color_hsl). Then in 2017, [hex with
+alpha](https://caniuse.com/css-rrggbbaa) appeared. Last, only recently,
+[`hwb()`](https://caniuse.com/mdn-css_types_color_hwb) started getting support
+in browsers. **All of these color spaces reference color within the same gamut,
+sRGB.**
+
+#### HEX
+
+{% Aside %}
+Colors are in the `sRGB Gamut`; 35.9% of the visible spectrum.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:261]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 25]
```diff
@@ -0,0 +1,1511 @@
...
+{% endAside %}
+
+<!-- {% BrowserCompat 'css.types.color.rgb_hexadecimal_notation' %} -->
+
+The hex colorspace specifies R, G, B and A with
+[hexadecimal](https://en.wikipedia.org/wiki/Hexadecimal). The following code
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:267]: fixed in doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 26]
```diff
@@ -0,0 +1,1511 @@
...
+  /* classic */
+  --3-digits: #49b;
+  --6-digits: #4095bf;
+
+  /* hex with opacity */
+  --4-digits-opaque: #f9bf;
+  --8-digits-opaque: #4095bfff;
+  --4-digits-with-opacity: #49b8;
+  --8-digits-with-opacity: #4095bf80;
+}
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:282]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 27]
```diff
@@ -0,0 +1,1511 @@
...
+  --modern: hwb(200deg 25% 25%);
+  --modern2: hwb(200 25% 25%);
+
+  --modern-with-opacity-percent: hwb(200 25% 25% / 50%);
+  --modern-with-opacity-decimal: hwb(200 25% 25% / .5);
+
+  /* hueless and no saturation */
+  --empty-channels-white: hwb(none 100% none);
+  --empty-channels-black: hwb(none none 100%);
+}
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:374]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 28]
```diff
@@ -0,0 +1,1511 @@
...
+```
+
+### Meet the new web color spaces
+
+Except for `sRGB` and `sRGB-linear`, the following color spaces offer access to
+larger gamuts than sRGB. The display-p3 color space offers almost twice as many
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:380]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 29]
```diff
@@ -0,0 +1,1511 @@
...
+colors as RGB while Rec2020 offers almost twice as many as display-p3. **That's a
+lot of colors!**
+
+{% Img
+  src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/BsVdm6fsHO8B0mBgkHQF.png",
+  alt="Five stacked triangles of varying color to help illustrate
+  the relationship and size of each of the new color spaces.",
+  width="800",
+  height="648"
+%}
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:390]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 30]
```diff
@@ -0,0 +1,1511 @@
...
+
+#### The `color()` function
+
+{% Aside %}
+Colors can be in the `sRGB Gamut`, `Display P3 Gamut`, `Rec2020 Gamut`; Up to 77.6% of the visible spectrum.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:395]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 31]
```diff
@@ -0,0 +1,1511 @@
...
+{% endAside %}
+
+<!-- {% BrowserCompat 'css.types.color.color' %} -->
+
+Consider this function a [higher order
+function](https://en.wikipedia.org/wiki/Higher-order_function) for any color
+spaces that specify colors with R, G and B channels; **exclusively a function
+for RGB**.
+[`color()`](https://developer.mozilla.org/docs/Web/CSS/color_value/color)
+takes a color space parameter first, then a series of channel values for RGB and
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:405]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 32]
```diff
@@ -0,0 +1,1511 @@
...
+optionally some alpha.
+
+**Pros**
+- A normalized space for accessing color spaces that use RGB channels.
+- Can scale up to any wide gamut RGB based color space.
+
+**Cons**
+- Only useful for RGB color spaces.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:413]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 33]
```diff
@@ -0,0 +1,1511 @@
...
+
+**Pros**
+
+- Nearly all displays support the range of this color space.
+- Design tool support.
+
+**Cons**
+
+- Not perceptually linear
+- Gradients often go through a dead zone (to be covered later in this article).
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:451]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 34]
```diff
@@ -0,0 +1,1511 @@
...
+```css
+.valid-css-srgb-colors {
+  --percents: color(srgb 34% 58% 73%);
+  --decimals: color(srgb .34 .58 .73);
+
+  --percents-with-opacity: color(srgb 34% 58% 73% / 50%);
+  --decimals-with-opacity: color(srgb .34 .58 .73 / .5);
+
+  /* empty */
+  --empty-channels-black: color(srgb none none none);
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:462]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 35]
```diff
@@ -0,0 +1,1511 @@
...
+  --percents: color(srgb-linear 34% 58% 73%);
+  --decimals: color(srgb-linear .34 .58 .73);
+
+  --percents-with-opacity: color(srgb-linear 34% 58% 73% / 50%);
+  --decimals-with-opacity: color(srgb-linear .34 .58 .73 / .5);
+
+  /* empty */
+  --empty-channels-black: color(srgb-linear none none none);
+  --empty-channels-black2: color(srgb-linear);
+}
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:502]: totally, will def make a gradient comparison 👍🏻

[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:502]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 36]
```diff
@@ -0,0 +1,1511 @@
...
+```
+
+#### LCH
+
+{% Aside %}
+Colors are in the `CIE Gamut`; 100% of the visible spectrum.
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:508]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 37]
```diff
@@ -0,0 +1,1511 @@
...
+{% endAside %}
+
+<!-- {% BrowserCompat 'css.types.color.lch' %} -->
+
+The first space of this post to introduce syntax for accessing colors outside
+the RGB gamut! It is also the first to make it very easy to create out of gamut
+color. This is because any CIE space color is capable of representing the entire
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:515]: yep def! thx!

File: site/en/blog/high-definition-css-color-guide/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:1001]: moved to doc

File: site/en/blog/high-definition-css-color-guide/index.md [hunk 38]
```diff
@@ -0,0 +1,1458 @@
...
+
+  <figcaption>
+    Preview for yourself: <a href="https://codepen.io/web-dot-dev/pen/poZgXxy">
+    https://codepen.io/web-dot-dev/pen/poZgXxy</a>
+  </figcaption>
+</figure>
+
+In supporting browsers, there's 50% more colors to pick from! You thought 16
+million colors sounded like a lot, wait until you see how many colors some of
+these new spaces can show. Also, think about all those gradients that
```
[the remark this answers is not in the document]
Comment (argyleink) [reply site/en/blog/high-definition-css-color-guide/index.md:90]: moved to the doc

**Review by argyleink — COMMENTED** [review]
All comments moved to the Google doc 🙂 Thanks everyone.

PR will receive an update once the document is approved again 👍🏻

File: site/en/blog/meet-the-new-css-color-spaces/index.md [hunk 39]
```diff
@@ -0,0 +1,36 @@
...
+subhead: >
+  CSS Color 4 brings wide gamut color tools and capabilities to the web: more colors, manipulation functions, and better gradients.
+date: 2023-02-01
+authors:
+  - argyle
+tags:
+  - css
+hero: image/vS06HQ1YTsbMKSFTIPl2iogUQP73/1EFisnQsVHucCNJdjQvy.png
+alt: >
+  A moment from The Wizard of Oz movie and half of it is in black and white and the other half is in color.
```
Comment (rachelandrew) [inline site/en/blog/meet-the-new-css-color-spaces/index.md:15]: I think you are using a different image for the blog

File: site/en/blog/meet-the-new-css-color-spaces/index.md [hunk 40]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+CSS color 4 brings a large set of tools and features to CSS for managing and
+handling color. I've written an article called the [High Definition Color
+Guide](articles/high-definition-css-color-guide/) to cover all these new
+features:
```
Comment (rachelandrew) [inline site/en/blog/meet-the-new-css-color-spaces/index.md:21]:
```suggestion
CSS color 4 brings a large set of tools and features to CSS for managing and
handling color. I've written the [High Definition Color
Guide](/articles/high-definition-css-color-guide/) to cover all these new
features. In this guide you will learn:
```

File: site/en/blog/meet-the-new-css-color-spaces/index.md [hunk 41]
```diff
@@ -0,0 +1,36 @@
...
+- A review of the classic color spaces
+- Meet the new web color spaces
+- Color interpolation
+- Gamut clamping
+- Choosing a color space
+- Migrating to HD CSS color
+- Checking for gamut and color space support
+- Debugging color with Chrome DevTools
+
+I hope this finds you well and answers all your color questions 🙂
```
Comment (rachelandrew) [inline site/en/blog/meet-the-new-css-color-spaces/index.md:36]:
```suggestion
- What is a color gamut?
- Human visual gamut.
- What is a color space?
- How to access more colors, new spaces, and debug results.
- A review of the classic color spaces.
- Meet the new web color spaces.
- Color interpolation.
- Gamut clamping.
- Choosing a color space.
- Migrating to HD CSS color.
- Checking for gamut and color space support.
- Debugging color with Chrome DevTools.

I hope this finds you well and answers all your color questions. 🙂
```

File: site/en/articles/high-definition-css-color-guide/index.md [hunk 42]
```diff
@@ -0,0 +1,1682 @@
...
+### Human visual gamut
+
+Color gamuts are often compared against the human visual gamut; the entirety of
+color we believe the human eye can see. HVS is often portrayed with a
+[chromaticity diagram](https://en.wikipedia.org/wiki/Chromaticity), like this:
+
+<figure>
+  {% Img
+    src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/98CgP4ILYvc0vOVyw2bp.png",
+    alt="todo",
```
Comment (rachelandrew) [inline site/en/articles/high-definition-css-color-guide/index.md:192]: Add alt text

**Review by rachelandrew — APPROVED** [review]
Approved so as not to be blocking, but there's a couple of alt text nits along with a couple of edits. Please check those before merging.

File: site/en/blog/meet-the-new-css-color-spaces/index.md (hunk 39 above)
Comment (argyleink) [reply site/en/blog/meet-the-new-css-color-spaces/index.md:15]: got it thx!

**Pull request merged**